component/core/src/main/java/org/exoplatform/social/core/jpa/storage/entity/SpaceEntity.java

```diff
@@ -22,35 +22,33 @@
 import java.util.HashSet;
 import java.util.Set;
 
-import javax.persistence.CascadeType;
-import javax.persistence.CollectionTable;
-import javax.persistence.Column;
-import javax.persistence.ElementCollection;
-import javax.persistence.Entity;
-import javax.persistence.FetchType;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.JoinColumn;
-import javax.persistence.NamedQueries;
-import javax.persistence.NamedQuery;
-import javax.persistence.OneToMany;
-import javax.persistence.SequenceGenerator;
-import javax.persistence.Table;
-import javax.persistence.Temporal;
-import javax.persistence.TemporalType;
+import javax.persistence.*;
 
 import org.exoplatform.commons.api.persistence.ExoEntity;
 
 @Entity(name = "SocSpaceEntity")
 @ExoEntity
 @Table(name = "SOC_SPACES")
+@NamedNativeQueries({
+        @NamedNativeQuery(
+                name = "SpaceEntity.getCommonSpacesBetweenTwoUsers",
+                query = "SELECT * FROM SOC_SPACES t " +
+                        "            where t.SPACE_ID in ( " +
+                        "            SELECT distinct t1.SPACE_ID FROM SOC_SPACES_MEMBERS t1 " +
+                        "            where t1.USER_ID = :user1 " +
+                        "            and (SELECT distinct t2.SPACE_ID FROM SOC_SPACES_MEMBERS t2 where t1.SPACE_ID=t2.SPACE_ID and t2.USER_ID = :user2 ) " +
+                        "            )",
+                resultClass = SpaceEntity.class
```

Comment:
I think resultClass is not needed

Reply:
In addition, can you put the query in NamedQueries instead of NativeNamedQueries

Reply:
in fact, it throws UndeclaredThrowableException because HQL did not support join queries so i had to use SQL query type, in addition resultClass is required when using NamedNativeQueries

Reply:
As you can see here :
https://github.com/exoplatform/agenda/blob/develop/agenda-services/src/main/java/org/exoplatform/agenda/entity/EventEntity.java#L85
It is possible to have a JOIN in a NamedQuery
I think you should rework a little your sql query and use a NamedQuery